k4EDM4hep2LcioConv/include/k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.h

```diff
@@ -159,12 +159,43 @@ std::unique_ptr<podio::CollectionBase> fillSubset(EVENT::LCCollection* LCCollect
 template <typename LCVecType>
 std::vector<CollNamePair> convertLCVec(const std::string& name, EVENT::LCCollection* LCCollection);
 
+/**
+ * Helper struct to wrap the functionality of putting parameters into a Frame in
+ * a semi type-erases way, such that it can easily be used as a template
+ * parameter
+ */
+struct ParamFramePutter {
+  ParamFramePutter() = delete;
+  ParamFramePutter(podio::Frame& f) : m_frame(f) {}
+
+  template <typename T>
+  void operator()(const std::string& key, const T& value) {
+    m_frame.putParameter(key, value);
+  }
+
+private:
+  podio::Frame& m_frame;
+};
+
+/**
+ * Converting all parameters of an LCIO Object and passing them to the PutParamF
+ * function that takes care of storing them appropriately.
+ *
+ * The indirection is necessary for better integration with k4FWCore where
+ * direct access to a Frame is not possible, but a putParameter method is
+ * available instead.
+ */
+template <typename LCIOType, typename PutParamF = ParamFramePutter>
+void convertObjectParameters(LCIOType* lcioobj, PutParamF putParamFun);
+
 /**
  * Converting all parameters of an LCIO Object and attaching them to the
  * passed podio::Frame.
  */
 template <typename LCIOType>
-void convertObjectParameters(LCIOType* lcioobj, podio::Frame& event);
+void convertObjectParameters(LCIOType* lcioobj, podio::Frame& event) {
+  convertObjectParameters(lcioobj, ParamFramePutter{event});
```

Comment:
```suggestion
    auto func = [&event](const std::string& key, const auto& value) {
      event.putParameter(key, value);
  };
  convertObjectParameters(lcioobj, func);
```
The `ParamFramePutter` struct can be not used (so it can be deleted) by using a lambda

Reply:
I have deliberately put it in to make it available downstream, e.g. in https://github.com/key4hep/k4MarlinWrapper/pull/216. But we can also remove it and repeat the lambda in two places. I don't have a strong opinion on that.

Reply:
But the lambda wouldn't need to be defined in the wrapper because if there is a frame this overload can be called. Only when there isn't a frame available then the lambda with `k4FWCore::{put,get}...` has to be defined, but that has to be defined anyway with `ParamFramePutter` or with this change.

Reply:
Ah yes, you are right. Didn't see the forest for the trees here.